src/pages/AccountManagementPage.tsx

```diff
@@ -0,0 +1,227 @@
+  const navigate = useNavigate()
+
+  function resetPassword(password) {
+    const oob = searchParams.get('oobCode')
+    fetch(buildPath('/doResetPassword'), {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ oobCode: oob, password: password }),
+    })
+      .then((res: Response) => {
+        if (res.ok) {
+          navigate('/')
+          return res.json()
+        }
+
+        setHasInvalidCredentials(true)
+        throw Error(res.statusText)
+      })
+      .catch((error: Error) => {
+        console.log('Password Reset Failed: ' + error.message)
+      })
+  }
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      resetPassword(password)
+    }
+  }
+
+  return (
+    <Box sx={{ p: 10 }}>
+      <Container maxWidth="md">
+        <Paper
+          sx={{
+            p: 10,
+            m: 5,
+          }}
+        >
+          <Stack spacing={3} alignItems={'center'}>
+            <Typography variant="h3">Reset Password</Typography>
+            <TextField
+              fullWidth
+              label="New Password"
+              variant="standard"
+              onChange={(e) => setPassword(e.target.value)}
+              onKeyDown={(e) => handleKeyPress(e)}
+            ></TextField>
+            <Box>
+              <Typography
+                fontSize={'18px'}
+                color={'red'}
+                visibility={hasInvalidCredentials ? 'visible' : 'hidden'}
+              >
+                Password is too weak. Please use more than 4 characters.
+              </Typography>
+            </Box>
+            <StylishButton
+              variant="outlined"
+              onClick={() => resetPassword(password)}
+            >
+              Submit
+            </StylishButton>
+          </Stack>
+        </Paper>
+      </Container>
+    </Box>
+  )
+}
+
+export default function AccountManagementPage() {
+  const [mode, setMode] = useState<string>('')
+  const [hasValidOob, setHasValidOob] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+
+  const [searchParams] = useSearchParams()
+  const user = useSelector((state: RootState) => state.user)
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    const urlMode = searchParams.get('mode')
+    const oob = searchParams.get('oobCode')
+
+    if (urlMode === 'resetPassword') {
+      fetch(buildPath('/checkResetPassword'), {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ oobCode: oob }),
+      })
+        .then((res: Response) => {
+          if (res.ok) {
+            setHasValidOob(true)
+            setIsLoading(false)
+            return res.json()
+          }
+
+          throw Error(res.statusText)
+        })
+        .catch(() => {
+          setIsLoading(false)
+          setHasValidOob(false)
+        })
+    } else {
```

Comment:
since you are deciding the mode based on url params, would it be better to explicitly say the else if verfiyEmail here? because it could call the api if url have different mode other than resetPassword

Reply:
True, I'll navigate them home if any other mode is used.